warehouse/sponsors/models.py

```diff
@@ -0,0 +1,38 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from sqlalchemy import Boolean, Column, String, sql
+from sqlalchemy.types import PickleType
+
+from warehouse import db
+from warehouse.utils.attrs import make_repr
+
+
+class Sponsor(db.Model):
+    __tablename__ = "sponsors"
+    __repr__ = make_repr("name")
+
+    name = Column(String, nullable=False)
+    service = Column(String)
+    # activity should be a list of strings
+    activity = Column(PickleType, nullable=False)
+
+    url = Column(String, nullable=False)  # TODO use URL field
+    color_logo_url = Column(String, nullable=False)  # TODO use URL field
+    white_logo_url = Column(String)  # TODO use URL field
```

Comment:
`sqlalchemy-utils` lib implements a [URLType](https://sqlalchemy-utils.readthedocs.io/en/latest/data_types.html#module-sqlalchemy_utils.types.url) that we can use here. But the [requirements structure](https://github.com/pypa/warehouse/tree/main/requirements) is not familiar to me and I don't know how to add the lib as a dependency. Can I have some help with that?

Reply:
https://warehouse.readthedocs.io/development/patterns.html#adding-new-dependencies

Reply:
Thanks! I've probably skiped this part from the docs =S

Reply:
At first I was iffy about storing this information as a URL, since we will be accepting uploads.

However, now I see that this is indeed the best way to go! Once we have an object store configured to accept logo uploads, we can have the Form/View in admin accept an image upload, immediately place it in the object store and populate the URL fields. This reduces having to implement a "FileField"/"ImageField" like Django has... and is only minorly problematic if we ever change object stores.

One note: We should be displaying sponsor logos [via camo](https://github.com/pypa/warehouse/blob/7fc3ce5bd7ecc93ef54c1652787fb5e7757fe6f2/warehouse/filters.py#L66-L85), if we are not already, rather than adding an additional [image-src CSP entry](https://github.com/pypa/warehouse/blob/dffce426d48d779f5495c01547234993e3a3e523/warehouse/csp.py#L97-L102).

Reply:
> One note: We should be displaying sponsor logos via camo, if we are not already, rather than adding an additional image-src CSP entry.

Are we planning to host the images ourselves, or are sponsors going to be responsible for hosting them?

Reply:
we'll be hosting them ourselves, but throwing them in a public object storage bucket.

that url will be gross, and honestly an unnecessary addition to CSP since we can camoify them (which has the added benefit of making all this work locally)

Reply:
👍